R/impact_central.R

```diff
@@ -152,22 +152,16 @@ impact_by_year_of_vaccination <- function(meta1, raw_impact, fvps, fvps_updates
     fvps_updates <- fvps_updates[fvps_updates$year %in% vaccination_years, ]
   }
 
-
+  raw_impact$birth_cohort <- raw_impact$time
   if (method == "method2a"){
-    if (vaccine_delivery$activity_type[1L] == "routine"){
-      raw_impact <- raw_impact[raw_impact$time %in% (vaccination_years - min(fvps$age)), ]
-    } else {
-      raw_impact <- raw_impact[raw_impact$time >= min(vaccination_years), ]
-    }
-    tot_impact <- stats::aggregate(value ~ country + burden_outcome, raw_impact, sum, na.rm = TRUE)
-    tot_fvps <- stats::aggregate(fvps~country, fvps, sum, na.rm = TRUE)
-    d <- merge_by_common_cols(tot_impact, tot_fvps, all = TRUE)
+    d <- impact_by_year_of_vaccination_country_perspective(
+      raw_impact, fvps, vaccine_delivery$activity_type[1L], vaccination_years)
   } else if (method == "method2b"){
-    cohort_impact <- raw_impact[raw_impact$time %in% (min(fvps$time):max(fvps$time)), c("country", "time", "burden_outcome", "value")]
-    cohort_fvps <- stats::aggregate(fvps ~ country + time, fvps, sum, na.rm = TRUE)
-    d <- merge_by_common_cols(cohort_impact, cohort_fvps, all = TRUE)
+    d <- impact_by_year_of_vaccination_cohort_perspective(raw_impact, fvps,
+                                                          vaccination_years)
   }
-  d$impact_ratio <- d$value / d$fvps
+  d$time <- d$birth_cohort
+  d$birth_cohort <- NULL
   d$value <- NULL
```

Comment:
Once you've had a chance to check this looks okay I will remove `value` and `fvps` from return of the `imapct_by_year*` functions but leaving it in now as the seeing which numbers are coming through makes it a bit easier to see what those functions are doing

Reply:
I would be happy to keep fvps in general- makes it easier to see what is going on even if it is unnecessary